[Feat] 지도 발자국 표시를 위한 일기 조회 API, 최신 일기 반환 반영

yoonsu417 opened the pull request:
## #️⃣연관된 이슈
> #50 

## 📝작업 내용
> 지도 화면에 들어갔을 때 발자국 표시를 위해 모든 일기를 리스트로 반환합니다.

## 🔎코드 설명
> DiaryController: DTO -> PathVariable 변경
> DiaryConverter:
    Diaries -> FootprintDiaryDTO 변환해주는 footprintDiaryDTO, 
    FootprintDiaryDTO -> FootprintDiaryListDTO로 변환해주는 footprintDiaryListDTO 추가
> DiaryQueryService, DiaryQueryServiceImpl: 조회 비즈니스 로직 작성
> DiaryRepository: Users로 해당하는 일기 찾기

## 💬고민사항 및 리뷰 요구사항 (Optional)
> 이 다음 마지막 지도 조회 api에서
> 1. 프론트에서 diaryId를 받아서 서버에서 clusterId 같은 애들 보내주기
> 2. 프론트에서 clusterId를 받아서 서버에서 같은 애들 보내주기 
두 개 중 어떤 걸 할지 고민중임
현재 발자국 찍을 때 diaryId와 clusterId 둘다 response 값에 있습니다 !
<img width="236" alt="스크린샷 2025-02-01 오후 11 05 42" src="https://github.com/user-attachments/assets/f5024518-15ae-409f-9ad7-d95f08c0991d" />

## 비고 (Optional)
> umc 워크북 9주차


## ⭐️수정⭐️ clusterId 이용한 발자국 찍기
- 처음 일기 작성 시 위도 경도 값을 비교하여
- if 동일한 장소로 파악되는 일기가 존재한다 ! -> 존재하는 일기와 같은 clusterId 부여
- else 동일한 장소 없다 ! if 근데 다른 장소는 있다 ! -> clusterId중 가장 큰 값 + 1 부여
- 동일한 장소도 다른 장소도 없다! -> 0 부여 로 구현을 했습니다 :)

## DB
- 첫 작성한 일기(id:26) -> 0 부여
- 첫 작성한 일기와 같은 장소(일기 id: 27,28,29) -> 일기 id 26과 같은 clusterId인 0부여
- 다른 장소(일기 id:30) -> 0+1 해서 1부여
- 다른 장소(일기 id:31) -> 일기 id 30과 같은 1 부여
- ... 등등
<img width="747" alt="스크린샷 2025-02-01 오후 11 17 10" src="https://github.com/user-attachments/assets/6277f720-b3ef-41a0-a5ea-439033e4e0d2" />



## Swagger
- 발자국 조회 시 최근에 쓴 일기로 반환
<img width="268" alt="스크린샷 2025-02-01 오후 11 13 49" src="https://github.com/user-attachments/assets/0ffc2ce0-aff4-4e98-a793-67258774dae7" />
<img width="242" alt="스크린샷 2025-02-01 오후 11 14 15" src="https://github.com/user-attachments/assets/62750fbd-6476-4592-8641-a6472fd839d7" />

yoonsu417 committed:
feat: #50 일기 발자국 표시 DTO 추가

requestDTO: 유저 아이디
responseDTO: 일기 list

yoonsu417 committed:
feat: #50 일기 발자국 조회 controller 기본 세팅

API URL, Request Body, Operation, return null

yoonsu417 committed:
feat: #50 일기 발자국 조회 기능 구현

DiaryController: DTO -> PathVariable 변경
DiaryConverter: Diaries -> FootprintDiaryDTO 변환해주는 footprintDiaryDTO, FootprintDiaryDTO -> FootprintDiaryListDTO로 변환해주는 footprintDiaryListDTO 추가
DiaryQueryService, DiaryQueryServiceImpl: 조회 비즈니스 로직 작성
DiaryRepository: Users로 해당하는 일기 찾기

Jinho622 commented on src/main/java/TtattaBackend/ttatta/converter/DiaryConverter.java:
```diff
@@ -33,6 +35,25 @@ public static DiaryPhotos toDiaryPhoto(String pictureUrl) {
                 .build();
     }
 
+    public static DiaryResponseDTO.FootprintDiaryDTO footprintDiaryDTO(Diaries diaries) {
```
메소드명 앞에 to를 붙여주세요!

yoonsu417 replied:
@Jinho622 수정 완료 했습니다 :)

Jinho622 commented on src/main/java/TtattaBackend/ttatta/converter/DiaryConverter.java:
```diff
@@ -33,6 +35,25 @@ public static DiaryPhotos toDiaryPhoto(String pictureUrl) {
+                .build();
+    }
+
+    public static DiaryResponseDTO.FootprintDiaryListDTO footprintDiaryListDTO(List<Diaries> diariesList) {
```
여기도 마찬가지로 메소드명 앞에 to를 붙여주세요!

yoonsu417 replied:
@Jinho622 완료했습니다 :)

yoonsu417 committed:
refactor: #50 일기 발자국 수정

Converter에 to 추가

yoonsu417 committed:
Merge remote-tracking branch 'origin/develop' into feature/#50-diary-footprint

# Conflicts:
#	src/main/java/TtattaBackend/ttatta/repository/DiaryRepository.java
#	src/main/java/TtattaBackend/ttatta/web/dto/DiaryResponseDTO.java

yoonsu417 committed:
refactor: #50 일기 발자국 최신 일기 반환

소수점 4자리까지 동일한 위도 경도를 같은 장소로 판단하고 거기서 가장 최신의 날짜 반환...

yoonsu417 committed:
refactor: #50 일기 위도 경도 소수점 5자리 수정

위도 경도로 일기 조회 시 소수점 5자리까지 같으면 같은 위치로 지정

yoonsu417 committed:
feat: #50 클러스터 시도중...

일기 작성 시 위도 경도 비교 후 클러스터 id 할당 지정 중

yoonsu417 committed:
feat: #50 일기 작성 시 클러스터 ID 할당

일기 작성 시 위도 경도 비교 후 같은 장소로 판단되면 해당 클러스터 id 부여, 다른 장소로 판단되면 가장 최근 clusterId + 1, 첫 일기면 clusterId 0으로 할당

yoonsu417 committed:
refactor: #50 일기 발자국 조회 시 클러스터 id 이용으로 수정

일기 발자국 조회 시 클러스터 id가 동일한 일기 중 가장 최근 일기로 반환

yoonsu417 committed:
Merge remote-tracking branch 'origin/develop' into feature/#50-diary-footprint

# Conflicts:
#	src/main/java/TtattaBackend/ttatta/converter/DiaryConverter.java
#	src/main/java/TtattaBackend/ttatta/repository/DiaryRepository.java
#	src/main/java/TtattaBackend/ttatta/service/DiaryService/DiaryCommandService.java
#	src/main/java/TtattaBackend/ttatta/service/DiaryService/DiaryQueryService.java
#	src/main/java/TtattaBackend/ttatta/service/DiaryService/DiaryQueryServiceImpl.java
#	src/main/java/TtattaBackend/ttatta/web/controller/DiaryController.java

yoonsu417 committed:
refactor: #50 일기 발자국 controller 수정

userId 토큰 처리 수정

yoonsu417 committed:
refactor: #50 createdAt date로 수정

createdAt date로 수정
